### PR TITLE
docs(migrate): add migration docs and update README

### PR DESCRIPTION
**Pull request by joshblack**
Closes IBM/carbon-components-react#1992

Add initial migration docs and update README.md

#### Changelog

**New**

- `docs/migration`

**Changed**

- `README.md` and `.github/CONTRIBUTING.md` content has been updated to reflect other package READMEs, also includes migration guides

**Removed**

### CODE REVIEW

**Comment by joshblack** [comment]
A note to reviewers, I'm not sure all the breaking changes in React so feel free to edit with content updates!

**Review by lovemecomputer — APPROVED** [review]
cool!

**Review by dakahn — APPROVED** [review]
Nice! 🔵

File: docs/migration/migrate-to-10.x.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+| `Link`                | No breaking changes                                                 |
+| `ListBox`             | No breaking changes                                                 |
+| `ListItem`            | No breaking changes                                                 |
+| `Loading`             | No breaking changes                                                 |
+| `Modal`               | No breaking changes                                                 |
+| `ModalWrapper`        | No breaking changes                                                 |
+| `MultiSelect`         | No breaking changes                                                 |
+| `Notification`        | Deprecated, use `InlineNotification` or `ToastNotification` instead |
+| `NumberInput`         | No breaking changes                                                 |
+| `OverflowMenu`        | No breaking changes                                                 |
```
Comment (asudoh) [inline docs/migration/migrate-to-10.x.md]:
```suggestion
| `OverflowMenu`        | [Migrate](../../src/components/OverflowMenu/migrate-to-10.x.md)     |
```

File: docs/migration/migrate-to-10.x.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+| `TextInput`           | No breaking changes                                                 |
+| `Tile`                | No breaking changes                                                 |
+| `TileGroup`           | No breaking changes                                                 |
+| `TimePicker`          | No breaking changes                                                 |
+| `TimePickerSelect`    | No breaking changes                                                 |
+| `Toggle`              | No breaking changes                                                 |
+| `ToggleSmall`         | No breaking changes                                                 |
+| `Toolbar`             | No breaking changes                                                 |
+| `ToolbarSearch`       | No breaking changes                                                 |
+| `Tooltip`             | No breaking changes                                                 |
```
Comment (asudoh) [inline docs/migration/migrate-to-10.x.md]:
```suggestion
| `Tooltip`             | [Migrate](../../src/components/Tooltip/migrate-to-10.x.md)          |
```

**Review by asudoh — COMMENTED** [review]
LGTM basically, some comments mostly upon the latest components state - Thanks @joshblack!

A discussion BTW would be whether we just deprecate legacy icon for `v10` or remove in `v10`. The former approach allows consumer to use alternate icon with legacy icon data structure, but may cause a false impression that usage of our legacy icon in `v10` is an accepted UX pattern.

**Pull request merged**